nodejs/eks/nodegroup.ts

```diff
@@ -755,35 +755,40 @@ ${customUserData}
         nodeAssociatePublicIpAddress = args.nodeAssociatePublicIpAddress;
     }
 
-    const numeric = new RegExp("^d+$");
+    const numeric = new RegExp("^\\d+$");
 
-    if (args.nodeRootVolumeIops && args.nodeRootVolumeType !== "io1") {
-        throw new Error(
-            "Cannot create a cluster node root volume of non-io1 type with provisioned IOPS (nodeRootVolumeIops).",
-        );
-    }
+    // We need to wrap the validation in a pulumi.all as MLCs could supply pulumi.Output<T> or T.
+    pulumi
```

Comment:
I'm a little worried about this paradigm since the result of the pulumi.all(xs).apply(f) is not awaited anywhere but presumably it's awaited by the SDK.

IN addition, it changes the flow of the program to not fail fast but move forward and fall through all these checks even if some of them are failing. It will create error races, I think, which validator fails first. 

Looks like this approach is present in the codebase before the change so perhaps OK.

```typescript

    core.apply((c) => {
        if (c.nodeGroupOptions.nodeSecurityGroup && args.nodeSecurityGroup) {
            if (
                c.nodeSecurityGroupTags &&
                c.nodeGroupOptions.nodeSecurityGroup.id !== args.nodeSecurityGroup.id
            ) {
                throw new Error(
                    `The NodeGroup's nodeSecurityGroup and the cluster option nodeSecurityGroupTags are mutually exclusive. Choose a single approach`,
                );
            }
        }
    });
```

Have you tested the negative path, perhaps manually, where you trigger this error?

Reply:
Yes, the process in which we check these args may arise to some interesting edge cases, though we don't seem to have any reports of this currently. I'm not inclined to do a refactor within this PR since the scope is solely to just ensure we can handle pulumi.outputs being presented in the args and to resolve a P1.

I have done a quick manual validation of triggering the `The NodeGroup's nodeSecurityGroup and the cluster option nodeSecurityGroupTags are mutually exclusive. Choose a single approach` error and can hit it successfully during preview on a new stack. The concern you share regarding fail fast seems like it'd only occur when we are validating between a true Output that is unknown during preview/check. In that case, it wouldn't fail early. Since we're only validating against resource inputs I don't think it'd be a concern for not failing fast.